xdsl/irdl/operations.py

```diff
@@ -615,7 +598,7 @@ class _SuccessorFieldDef(_OpDefField[SuccessorDef]):
 
 
 def result_def(
-    constraint: (AttrConstraint | Attribute | type[Attribute] | TypeVar) = Attribute,
+    constraint: IRDLAttributeConstraint = Attribute,
```

Comment:
why are some of these generic and some not?

Reply:
I see that this isn't actually changed so maybe lets not address this here

Reply:
To answer this, when you do `op.my_attr`, the attribute will always be correct.
But if you do `op.my_operand`, the operand might not be of the correct type because you are in the middle of a rewriting.
That's why one should keep the generic types while the other not.

But that will change later once someone implement conversion patterns.